js_modules/dagster-ui/packages/ui-core/src/asset-graph/AssetGraphExplorerFilters.tsx

```diff
@@ -138,11 +179,21 @@ export function useAssetGraphExplorerFilters({
   if (assetGroups) {
     filters.push(groupsFilter);
   }
+
+  const {isBranchDeployment} = React.useContext(CloudOSSContext);
+  if (
+    changedInBranch &&
+    featureEnabled(FeatureFlag.flagExperimentalBranchDiff) &&
+    isBranchDeployment
+  ) {
+    filters.push(changedFilter);
+  }
```

Comment:
@hellendag This is the gating. The alternative to this seems way too complex because the filter values/callbacks need to be accessible a component parent components up and the values also need to be persisted to the URL which complicates things even more